app/jobs/store_version_contents_job.rb

```diff
@@ -0,0 +1,39 @@
+class StoreVersionContentsJob < ApplicationJob
+  include GoodJob::ActiveJobExtensions::Concurrency
+  good_job_control_concurrency_with total_limit: 1, key: -> { "store-contents-#{version_arg.full_name}" }
+  queue_as :default
+  self.queue_adapter = :good_job
+
+  class VersionNotIndexed < RuntimeError; end
+  class GemNotFound < RuntimeError; end
+
+  discard_on ActiveJob::DeserializationError
+  discard_on Gem::Package::FormatError, Gem::Security::Exception
+
+  retry_on VersionNotIndexed, wait: :exponentially_longer, attempts: 5
+  retry_on GemNotFound, wait: :exponentially_longer, attempts: 5
+
+  rescue_from(GemNotFound, Gem::Package::FormatError, Gem::Security::Exception) do |error|
+    version = version_arg.full_name
+    Rails.logger.error "Storing gem contents for #{version} failed: #{error.class}: #{error.message}"
```

Comment:
```suggestion
    logger.error "Storing gem contents for #{version} failed", error
```

should take advantage of the newly-merged semantic_logger